### ISSUE TITLE
export: ONNX contract hardcodes a single BCE logit, wrong for multiclass checkpoints

### ISSUE THREAD

**Issue by jayhesselberth** [rnabioco/leech#357]
## Bug

The ONNX export contract hardcodes the binary output convention. A multiclass checkpoint gets a sidecar that describes its output incorrectly.

`src/leech/model_export.py:204-212` (main @ 75232f6) always writes:

```json
{"name": "logits", "shape": ["batch", 1],
 "convention": "a SINGLE BCE logit, not a two-class softmax. P(positive) = sigmoid(logits)."}
```

The same claim appears in the `onnx_export.py` module docstring ("leech classifiers emit a single BCE logit `(N, 1)`").

## Repro

Export a checkpoint trained with `num_out: 22`, `loss_type: cross_entropy` (the 22-way amino-acid model, `2026-aa-trna-models/results/noncognate-22way/TCNDwellResidualLN/multiclass/`). The graph's output is `[B, 22]`, but the `.json` contract says `[batch, 1]` and tells the consumer to apply `sigmoid`.

Found by two independent export runs for rnabioco/escapepod-models#144. The runs used the pinned leech 0.10.0, and the same code is on main. escapepod-models works around it by writing its own output block into the bundle, but any consumer that trusts leech's sidecar applies the wrong transform. Following the docstring's own warning, "every call is wrong and nothing errors."

## Expected

Derive the output block from the checkpoint config:
- `num_out == 1`: the current BCE-logit convention.
- `num_out > 1`: shape `[batch, num_out]`, softmax convention, and the class order from `label_map`.
- If a calibration is present, name it (`method`, e.g. `matrix`), since multiclass calibration is applied to logits before the softmax.

Also check the exported graph's output width against `num_out` at export time.

## Acceptance

- A test that exports a small multiclass model and asserts the contract's output shape and convention match the graph.
- The `onnx_export.py` docstring no longer says every classifier is binary.
